components/renku_data_services/resource_pool_adapters/adapter.py

```diff
@@ -309,23 +342,16 @@ async def update_resource_pool(self, api_user: base_models.APIUser, id: int, **k
                                         message="More fields than the id of the class "
                                         "should be provided when updating it"
                                     )
-                                stmt_cls = (
-                                    update(schemas.ResourceClassORM)
-                                    .where(schemas.ResourceClassORM.id == class_id)
-                                    .where(schemas.ResourceClassORM.resource_pool_id == id)
-                                    .values(**cls)
-                                    .returning(schemas.ResourceClassORM)
+                                new_cls = await self.update_resource_class(
```

Comment:
This seems like a nice place to use `asyncio.gather()` outside the loop without awaiting here

Reply:
Done.